### PR TITLE
Add support for github tokens for private repos.

### PR DESCRIPTION
**Pull request by Marenz**

### CODE REVIEW

**Comment by Marenz** [comment]
For anyone who wants to use this already, this branch provides a packaged dist: https://github.com/Marenz/check-for-changed-files/tree/token-support-packaged

 So you can use this with `marenz/check-for-changed-files/token-support-packaged`

File: action.yml [hunk 1]
```diff
@@ -19,6 +19,10 @@ inputs:
     description: "String template used when the check fails"
     required: false
     default: "the prerequisite file pattern ${prereq-pattern} matched changed files in the pull request, but the ${file-pattern} pattern did NOT and the ${skip-label} label is not set"
+  token:
+    description: "Github token to use to checkout the pull request files."
```
Comment (brettcannon) [inline action.yml:23]:
```suggestion
    description: "GitHub token to use to checkout the pull request files."
```

File: README.md [hunk 2]
```diff
@@ -41,6 +41,11 @@ The message to emit when the check fails. All other inputs can be specified in
 the message using `${}` syntax, e.g. `${file-pattern}` for the `file-pattern`
 input. All values will be quoted for easy identification of any whitespace.
 
+### `token`
+
+Github token to use for private repositories. Falls back to anonymous access if
```
Comment (brettcannon) [inline README.md:46]:
```suggestion
A GitHub auth token to use for private repositories. Falls back to anonymous access if
```

Is there a good URL to point users to for what sort of token to request?

**Review by brettcannon — CHANGES_REQUESTED** [review]
Very minor doc tweaks.

**Comment by brettcannon** [comment]
This change is failing CI due to not having the `index.js` file updated.

File: README.md (hunk 2 above)
Comment (llucax) [reply README.md:46]: In most cases probably using [`${{ secrets.GITHUB_TOKEN }}`](https://docs.github.com/en/actions/security-guides/automatic-token-authentication) should be enough.

**Comment by Marenz** [comment]
I addressed all comments and mentioned the github secrets that usually is used in that place

**Review by brettcannon — APPROVED** [review]

**Comment by brettcannon** [comment]
Thanks! I will work on getting a new release out.

**Pull request merged**